packages/kbn-dependency-tree/README.md

```diff
@@ -0,0 +1,110 @@
+# @kbn/dependency-tree
+
+A CLI tool for visualizing TypeScript dependency trees using `tsconfig.json` files with `kbn_references`.
+
+## Usage
+
+### Command Line Interface
+
+```bash
+node scripts/kbn_dependency_tree <tsconfig.json> [options]
+```
+
+#### Basic Examples
+
+```bash
+# Analyze ML plugin dependencies
+node scripts/kbn_dependency_tree x-pack/platform/plugins/shared/ml/tsconfig.json
+
+# Self-referential example - analyze this package's own dependencies
+node scripts/kbn_dependency_tree packages/kbn-dependency-tree/tsconfig.json
+
+# Limit depth to avoid deep traversal
+node scripts/kbn_dependency_tree packages/some-package/tsconfig.json --depth 2
+
+# Filter to only show ML-related dependencies
+node scripts/kbn_dependency_tree x-pack/platform/plugins/shared/ml/tsconfig.json --filter "@kbn/ml-"
+
+# Show package paths in output
+node scripts/kbn_dependency_tree packages/some-package/tsconfig.json --paths
+
+# Output as JSON for further processing
+node scripts/kbn_dependency_tree packages/some-package/tsconfig.json --json
+```
+
+#### Command Line Options
+
+| Option               | Description                             | Default |
+| -------------------- | --------------------------------------- | ------- |
+| `--depth <n>`        | Maximum depth to traverse               | `3`     |
```

Comment:
What's the expected output for `--depth 0`? It looks like it traverses all the way down but should it only show the first level of branches?